AzureMonitorAgent/agent.py

```diff
@@ -1146,8 +1146,22 @@ def find_package_manager(operation):
     for rpm_dist in rpm_set:
         if dist.startswith(rpm_dist):
             PackageManager = "rpm"
-            # Same as above.
+            # Same as above; need to force.
             PackageManagerOptions = "--force"
+
+            # If FIPS is enabled and RPM version is 4.14 or higher, we need to apply the below workaround
+            try:
+                fips_code, _ = run_command_and_log('grep -i "fips=1" /proc/cmdline 1>/dev/null 2>&1', check_error=False, log_cmd=False)
```

Comment:
does proc check work with older rpms? redhat 6 and 7? and newer onces like 8.*?

Reply:
I think it is good to confirm it returns the right enable flag for fips for each of rel families: 1 OS 6.*, 1 OS 7.* and 1 OS 8.*. I remember reading somewhere that the enabled flag for fips is not always returning the correct value.

Reply:
Confirmed that RHEL7 with FIPS enabled has `fips=1` in `/proc/cmdline` (also confirmed that AMA will install on RHEL7 w/FIPS with no `--nodigest` flag). RHEL6 series doesn't matter since we don't support.

Reply:
Do you want to hold on this change for now? Just realized that AMA does not support Centos 6 installation. In case of OMS upgrading rpm build to >4.14 means dropping Cent os/ rel 6 install which does not seem to be the case with AMA.

Reply:
I'm not sure how there's any interaction/relation between what we do with OMS and what we do with AMA here.

Reply:
In OMS we went with noDigest flag instead of creating FIPS compliant package for 2 reasons:
1.  Updating oms build system to centos 7 would mean dropping customers who are on redhat 6 or centos 6. In case of AMA , it is not an issue as these distros never been supported by AMA nor there are AMA customers that will break.
2.  OMS agent is on deprecation path

In case of AMA,  from FIPs compliance point of view, it might not be acceptable to customers to do same with new agent given #1 is not an issue for AMA.  Please follow up with PM and Dmitry to confirm they approve this decision. 

@Dmitry-Matveev